elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -443,9 +453,11 @@ export class PeerPool extends EventEmitter {
 
 		// Throw an error because adding a peer multiple times is a common developer error which is very difficult to identify and debug.
 		if (this._peerMap.has(peer.id)) {
-			throw new Error(`Peer ${peer.id} was already in the peer pool`);
+			this._peerMapDuplicates.set(peer.id, peer);
```

Comment:
Why don't we remove the peer straight away in here? I don't think we need the `_peerMapDuplicates` list. Could you explain why we need it?

Reply:
If we remove the peer that means we are removing the outbound connection of that peer and then only making inbound that will again create the same problem which we are trying to solve with this fix.